consensus/state.go

```diff
@@ -354,6 +355,29 @@ func (cs *State) OnStart() error {
 		return err
 	}
 
+	// Double Signing Risk Reduction
```

Comment:
is there any way we could get a testcase for this functionality?

Reply:
Yes, I'm writing it now, 
I'm trying to write an integration test using a docker like `test/p2p/fast_sync`. Is that okay?